test/hotspot/jtreg/compiler/lib/compile_framework/CompileFramework.java

```diff
@@ -0,0 +1,254 @@
+        for (SourceCode sourceCode : sourceCodes) {
+            System.out.println("SourceCode: " + sourceCode.className + " " + sourceCode.kind.name().toLowerCase());
+            System.out.println(sourceCode.code);
+        }
+    }
+
+    public void compile() {
+        if (classLoader != null) {
+            throw new RuntimeException("Cannot compile twice!");
+        }
+
+        printSourceCodes();
+
+        List<SourceCode> javaSources = new ArrayList<SourceCode>();
+        List<SourceCode> jasmSources = new ArrayList<SourceCode>();
+        for (SourceCode sourceCode : sourceCodes) {
+            switch (sourceCode.kind) {
+                case SourceCode.Kind.JASM -> { jasmSources.add(sourceCode);  }
+                case SourceCode.Kind.JAVA -> { javaSources.add(sourceCode);  }
+            }
+        }
+
+        String sourceDir = getSourceDirName();
+        compileJasmSources(sourceDir, jasmSources);
+        compileJavaSources(sourceDir, javaSources);
+        setUpClassLoader();
+    }
+
+    private static String getSourceDirName() {
+        // Create temporary directory for jasm source files
+        final String sourceDir;
+        try {
+            sourceDir = "compile-framework-sources-" + ProcessTools.getProcessId();
+        } catch (Exception e) {
+            throw new InternalCompileFrameworkException("Could not get ProcessID", e);
+        }
+        System.out.println("Source directory: " + sourceDir);
+        return sourceDir;
+    }
+
+    private static void compileJasmSources(String sourceDir, List<SourceCode> jasmSources) {
+        if (jasmSources.size() == 0) {
+            System.out.println("No jasm sources to compile.");
+            return;
+        }
+        System.out.println("Compiling jasm sources: " + jasmSources.size());
+
+        List<String> jasmFileNames = writeSourcesToFile(sourceDir, jasmSources);
+        compileJasmFiles(jasmFileNames);
+        System.out.println("Jasm sources compiled.");
+    }
+
+    private static void compileJasmFiles(List<String> fileNames) {
+        // Compile JASM files with asmtools.jar, shipped with jtreg.
+        List<String> command = new ArrayList<>();
+
+        command.add("%s/bin/java".formatted(System.getProperty("compile.jdk")));
+        command.add("-classpath");
+        command.add(getAsmToolsPath());
+        command.add("org.openjdk.asmtools.jasm.Main");
+        command.add("-d");
+        command.add(System.getProperty("test.classes"));
+        for (String fileName : fileNames) {
+            command.add(new File(fileName).getAbsolutePath());
+        }
+
+        executeCompileCommand(command);
+    }
+
+    private static String[] getClassPaths() {
+        String separator = File.pathSeparator;
+        return System.getProperty("java.class.path").split(separator);
+    }
+
+    private static String getAsmToolsPath() {
+        for (String path : getClassPaths()) {
+            System.out.println("jtreg.jar?: " + path);
```

Comment:
A weird debug printout here

Reply:
Ah, good point. Needed that for debugging on windows